docker_services_cli/env.py

```diff
@@ -63,22 +63,95 @@ def _load_or_set_env(services_version, default_version):
+                    env_var_with_version = (
+                        f"{service_name.upper()}_"
+                        f"{service_override_version}_LATEST"
+                    )
+                    if (SERVICES_ALL_DEFAULT_VERSIONS.get(
+                       env_var_with_version)):
+                        os.environ[
+                            f"{service_name.upper()}_VERSION"
+                        ] = SERVICES_ALL_DEFAULT_VERSIONS.get(
+                            env_var_with_version
+                        )
+                    else:
+                        available_major_versions = [
+                            v.split(".")[0]
+                            for v in SERVICES[service_name][
+                                "DEFAULT_VERSIONS"
+                            ].values()
+                        ]
+                        click.secho(
+                            f"No major version {service_override_version} "
+                            f"for {service_name}. "
+                            "Please use one of the available "
+                            f"ones: {available_major_versions}",
+                            fg="red",
+                        )
+                        exit(1)
+                    num_services_to_override -= 1
+                    if not num_services_to_override:
+                        return
+
 
 def set_env():
     """Export the environment variables for services and versions."""
-    # export variables for latest versions
-    for key, value in DEFAULT_VERSIONS.items():
+    for key, value in SERVICES_ALL_DEFAULT_VERSIONS.items():
         _set_default_env(key, value)
 
-    # export services configuration
-    for service in SERVICES:
+    for service in SERVICES.values():
         for key, value in service.items():
-            if key.endswith('_VERSION'):
+            if key.endswith("_VERSION"):
                 _load_or_set_env(key, value)
-            else:
-                _set_default_env(key, value)
+            elif key == "CONTAINER_CONFIG_ENVIRONMENT_VARIABLES":
+                for envvar_name, envvar_value in value.items():
+                    _set_default_env(envvar_name, envvar_value)
+
+
+def print_setup_env_config(services, called_from, env_set_command="export"):
+    """Prints setup environment instructions."""
+    should_print_instructions = False
+    for service in services:
+        for key, value in (
+            SERVICES.get(service)
+            .get("CONTAINER_CONNECTION_ENVIRONMENT_VARIABLES", {})
+            .items()
+        ):
+            command = f"{env_set_command} {key}"
+            if env_set_command == "export":
+                command += f"={value}"
+            click.echo(command)
+            should_print_instructions = True
+
+    if should_print_instructions:
+        click.secho("# Configure your environment running:", fg="yellow")
+        cmd_template = \
+            '# eval "$(docker-services-cli {cmd} {services_list} --env)"'
```

Comment:
Moderate: When I run `docker-services-cli down --env` in the output I get

```console
# Configure your environment running:
# eval "$(docker-services-cli down elasticsearch postgresql mysql redis memcached rabbitmq --env)"
```
However,  this `eval` command doesn't work as `down` doesn't receive any options, I think we should eliminate the services list when `cmd` is `down`.